### DISCUSSION TITLE
How do I link github with the correct microsoft account (which I already have)?

### DISCUSSION THREAD

**Discussion by fnmoura** [community/community#70280] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I linked the github account with a new microsoft account (I created the account wrongly).

How do I link github with the correct microsoft account (which I already have)?

**Comment by itsdotnet** [comment]
**Unlink the Wrong Microsoft Account:**

1. Log in to your GitHub account using the wrong Microsoft account.

2. Click on your profile picture in the top right corner and select "Settings."

3. In the left sidebar, click on "Applications."

4. Under "GitHub Apps," find the application linked with your Microsoft account and click on it.

5. Scroll down to the "Access" section, and under "Authorized," you should see the Microsoft account that is currently linked.

6. Click "Revoke" or "Uninstall" to unlink the wrong Microsoft account.

**Link the Correct Microsoft Account:**

Now that you've unlinked the wrong Microsoft account, you can link the correct one.

1. Log out of your GitHub account if you are still logged in using the wrong Microsoft account.

2. Go to GitHub and click on "Sign in."

3. Click on "Sign in with Microsoft."

4. You will be prompted to enter the credentials for your correct Microsoft account.

5. Once you've successfully logged in with the correct Microsoft account, your GitHub account will be linked to it.

**Comment by oussamaef06** [comment]
If you've linked your GitHub account with the wrong Microsoft account and want to link it with the correct Microsoft account, you can follow these steps:

Unlink the Wrong Microsoft Account:

To unlink your GitHub account from the wrong Microsoft account, you need to revoke access for the GitHub application in your Microsoft account settings. Here's how:

Go to your Microsoft account settings: https://account.live.com/consent/Manage
Sign in with the wrong Microsoft account.
Find and select the GitHub application (it should be listed under "Apps and services you've given access to").
Revoke access or remove the permission for GitHub.
Link GitHub with the Correct Microsoft Account:

After you've unlinked the wrong Microsoft account, you can link your GitHub account with the correct Microsoft account:

Log in to your GitHub account.
In the upper-right corner, click on your profile picture and go to "Settings."
In the left sidebar, click on "Applications."
Under "Authorized OAuth Apps," you should see "Microsoft" as one of the authorized apps.
Click on "Configure" next to "Microsoft."
You'll be prompted to sign in to your Microsoft account. Sign in with the correct Microsoft account that you want to link to your GitHub account.
Authorize the connection between your GitHub and Microsoft accounts.
Once you've completed these steps, your GitHub account will be correctly linked with your Microsoft account.

Please note that the specific steps may change, as GitHub and Microsoft may update their account linking procedures over time. If you encounter any issues during this process, consider reaching out to GitHub support or Microsoft support for assistance.

**Comment by Tbl0x7D6** [comment, marked as answer]
look at this:
https://learn.microsoft.com/en-us/visualstudio/subscriptions/sign-in-github

Go to the [Security tab](https://account.microsoft.com/security) of account.microsoft.com and select Advanced security options to unlink your GitHub account. Unlinking your GitHub account removes it as a sign-in method and removes access to any GitHub repositories in Visual Studio. Other Microsoft products might request access your GitHub account separately, so removing the access here doesn't remove access in all products. Go to the [application permissions](https://github.com/settings/applications) page of your GitHub profile to revoke consent from the apps listed there.

**Reply by lixiaoyu3** [reply]
Thank you very much for your help, most of the web about unlinking Microsoft account and github account is wrong, thank you for solving a big problem for me!

**Comment by deepaksinghrx** [comment]
look at this:
https://learn.microsoft.com/en-us/visualstudio/subscriptions/sign-in-github

Go to the [Security tab](https://account.microsoft.com/security) of account.microsoft.com and select Advanced security options to unlink your GitHub account. Unlinking your GitHub account removes it as a sign-in method and removes access to any GitHub repositories in Visual Studio. Other Microsoft products might request access your GitHub account separately, so removing the access here doesn't remove access in all products. Go to the [application permissions](https://github.com/settings/applications) page of your GitHub profile to revoke consent from the apps listed there.
